src/v/config/tls_config.h

```diff
@@ -228,26 +199,20 @@ struct convert<config::tls_config> {
         }
         auto enabled = node["enabled"] && node["enabled"].as<bool>();
         if (!enabled) {
-            rhs = config::tls_config(
-              false, std::nullopt, std::nullopt, false, std::nullopt);
+            rhs = config::tls_config(false, std::nullopt, std::nullopt, false);
         } else {
             auto key_cert
               = node["key_file"]
                   ? std::make_optional<config::key_cert>(config::key_cert{
                     to_absolute(node["key_file"].as<ss::sstring>()),
                     to_absolute(node["cert_file"].as<ss::sstring>())})
                   : std::nullopt;
-            auto principal_mapping_rules
-              = node["principal_mapping_rules"]
-                  ? node["principal_mapping_rules"].as<ss::sstring>()
-                  : std::optional<ss::sstring>();
             rhs = config::tls_config(
               enabled,
               key_cert,
               to_absolute(read_optional(node, "truststore_file")),
               node["require_client_auth"]
```

Comment:
require_client_auth i guess is what is passed into seastar to enable mtls? does this mean that users need to set this, or are we going to do that for them if they have mtls enabled?

Reply:
I don't think it's needed any more. I'll see if I can write a test.